universal-login-commons/lib/integration/ethereum/BalanceChecker.ts

```diff
@@ -12,12 +12,14 @@ export class BalanceChecker {
     return this.getTokenBalance(walletAddress, tokenAddress);
   }
 
-  private getEthBalance(walletAddress: string) {
-    return this.provider.getBalance(walletAddress);
+  private async getEthBalance(walletAddress: string) {
+    const balance = await this.provider.getBalance(walletAddress);
+    return balance.toString();
```

Comment:
Why not `BigNumber`?

Reply:
Because it complicates code down the road

Reply:
`toCanonical(value: BigNumber)`